src/typeorm-aurora-data-api-driver.ts

```diff
@@ -1,95 +1,113 @@
+      const parameter = parameters![numberOfParametersInQueryString]
+      if (Array.isArray(parameter)) {
+        parameter.forEach((_, index) => {
+          const isLastParameter = index === parameter.length - 1
+          const paramName = `:e_param_${extendedParameterCount}${isLastParameter ? '' : ', '}`
+          extendedParameterCount += 1
+          newQueryString += paramName
+        })
+      } else {
         const paramName = `:param_${numberOfParametersInQueryString}`
-
         numberOfParametersInQueryString += 1
         newQueryString += paramName
-      } else if (quoteCharacters.includes(currentCharacter) && !currentCharacterEscaped) {
-        if (!currentQuote) {
-          currentQuote = currentCharacter
-        } else if (currentQuote === currentCharacter) {
-          currentQuote = null
-        }
-
-        newQueryString += currentCharacter
-      } else {
-        newQueryString += currentCharacter
       }
+
+      continue
     }
 
-    if (
-      parameters &&
-      parameters.length > 0 &&
-      parameters.length % numberOfParametersInQueryString !== 0
-    ) {
-      throw new Error(`Number of parameters mismatch, got ${numberOfParametersInQueryString} in query string \
-            and ${parameters.length} in input`)
+    if (quoteCharacters.includes(currentCharacter) && !currentCharacterEscaped) {
+      if (!currentQuote) {
+        currentQuote = currentCharacter
+      } else if (currentQuote === currentCharacter) {
+        currentQuote = null
+      }
     }
 
-    const transformedParameters: any[] = []
+    newQueryString += currentCharacter
+  }
+
+  return [newQueryString, numberOfParametersInQueryString]
+}
+
+const transformParameters = (numberOfParametersInQueryString: number, parameters?: any[]) => {
+  if (
+    parameters &&
+    parameters.length > 0 &&
+    parameters.length % numberOfParametersInQueryString !== 0
+  ) {
+    throw new Error(
+      `Number of parameters mismatch, got ${numberOfParametersInQueryString} in query string \
+      and ${parameters.length} in input`)
+  }
 
-    if (parameters && parameters.length > 0) {
-      const numberOfObjects = parameters.length / numberOfParametersInQueryString
+  let extendedParameterCount = 0
+  const transformedParameters: any[] = []
 
-      for (let i = 0; i < numberOfObjects; i += 1) {
-        const parameterObject: any = {}
+  if (parameters && parameters.length > 0) {
+    const numberOfObjects = parameters.length / numberOfParametersInQueryString
 
-        for (let y = 0; y < numberOfParametersInQueryString; y += 1) {
-          const paramName = `param_${y}`
+    for (let i = 0; i < numberOfObjects; i += 1) {
+      const parameterObject: any = {}
 
-          parameterObject[paramName] = parameters[i + y]
-        }
+      for (let y = 0; y < numberOfParametersInQueryString; y += 1) {
+        const parameter = parameters[i + y]
 
-        transformedParameters.push(parameterObject)
+        Array.isArray(parameter)
```

Comment:
I'd rather have if-else clause here. Also remove the ignore and make i+= 1 please because it makes it easier to see

Reply:
Matter of preference, but sure no problemo

Reply:
if-else is actually definitely nicer in this case :)